### DISCUSSION TITLE
Why doesn't UTM support Apple's virtualization framework on Intel Macs?

### DISCUSSION THREAD

**Discussion by steven-michaud** [utmapp/UTM#4734] [General]
As best I can tell, even the latest UTM release doesn't support using Apple's virtualization framework on Intel Macs. It only does so on Apple Silicon Macs.

Why is this? Are there technical reasons? Or did you decide there's no practical need for this functionality? (Given that VMware and Parallels support full-featured VMs on Intel Macs.)

I'd like to use the virtualization framework on Intel Macs to debug problems and design flaws in the same framework on Apple Silicon Macs (notably https://github.com/utmapp/UTM/issues/4026). I have much better debugging tools on Intel -- for example [HookCase](https://github.com/steven-michaud/HookCase).

**Comment by osy** [comment]
Vmware and parallels spent billions of dollars developing their own solution. There’s no apple provided framework.

**Comment by steven-michaud** [comment]
> There’s no apple provided framework.

Not true. There's a `Virtualization.framework` in `/System/Library/Frameworks` on Intel Macs on both macOS 12 and 13.

But you've answered my question, at least indirectly. The reason UTM doesn't support the virtualization framework on Intel Macs is that there's no practical need for this functionality.

I'm going to try building UTM myself, altering it to use the virtualization framework on Intel Macs. I'll report my results here.

**Reply by osy** [reply]
Yeah I know the framework exists. You can actually use it on Intel UTM for running Linux. Read up on the apis and you’ll see.

**Comment by steven-michaud** [comment]
OK, so there are both practical and technical reasons -- presuming that Apple's virtualization framework doesn't support macOS clients on Intel Macs.

I'm still going to play around with it, and report my results here. It'd be useful to know the limits of Apple's virtualization framework on Intel Macs.

**Comment by steven-michaud** [comment]
> presuming that Apple's virtualization framework doesn't support macOS clients on Intel Macs

This seems to be true. In fact a whole bunch of Objective-C/Swift classes that are present in the arm64e Virtualization framework are missing in its x86_64 copy. Among those whose names start with "VZMac", only `VZMacGraphicsDisplayConfiguration` is present on both. Important missing classes, used by UTM, include `VZMacOSBootLoader`, `VZMacHardwareModel`, `VZMacPlatformConfiguration`, `VZMacOSVirtualMachineStartOptions`, `VZMacOSInstaller` and `VZMacOSRestoreImage`.

I wonder if Apple will ever fix this. I kind of doubt it.

**Comment by griffindavidson** [comment]
Apple's Virtualization framework _is_ supported on Intel Macs, but it was never ported to VM macOS on Intel machines, it works great with Linux however.

I presume it likely wasn't ported to create macOS VM's on Intel in a attempt to hinder Hackintosh development, such as [Adespoton's UTM Config files](https://github.com/adespoton/utmconfigs) and of course OpenCore
